src/main/java/org/spongepowered/api/data/manipulator/immutable/entity/ImmutableTextureData.java

```diff
@@ -25,27 +25,30 @@
 package org.spongepowered.api.data.manipulator.immutable.entity;
 
 import org.spongepowered.api.data.manipulator.ImmutableDataManipulator;
-import org.spongepowered.api.data.manipulator.mutable.entity.SkinData;
-import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.manipulator.mutable.entity.TextureData;
+import org.spongepowered.api.data.value.immutable.ImmutableOptionalValue;
 import org.spongepowered.api.entity.living.Humanoid;
-import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.profile.property.ProfileProperty;
 
 import java.util.UUID;
 
 /**
- * An {@link ImmutableDataManipulator} handling the {@link UUID} for  the
- * {@link Humanoid} skin used. Usually this is meant where the {@link UUID}
- * belongs to a {@link Player} but without relying on a {@link Player},
- * the {@link Humanoid} will use the same skin url on the server.
+ * An {@link ImmutableDataManipulator} for the textures of a {@link Humanoid}.
  */
-public interface ImmutableSkinData extends ImmutableDataManipulator<ImmutableSkinData, SkinData> {
+public interface ImmutableTextureData extends ImmutableDataManipulator<ImmutableTextureData, TextureData> {
```

Comment:
@JBYoshi Read the comment just above yours - this represents more than just the skin, and it's known as "textures" internally.